rust/src/read.rs

```diff
@@ -23,8 +19,10 @@ use enumset::{enum_set, EnumSet, EnumSetType};
 use log::*;
 
 use crate::{
-    io_utils::CountingCrcReader,
     records::{self, op, Record},
+    sans_io::read::{
```

Comment:
I don't love the name, sans_io::read seems like a oxymoron. I guess mcap is taken. Maybe mcap_reader or buffered_reader or something like that? Not sure, it's not a hill I will die on.